FirebaseInAppMessaging/Sources/FIRInAppMessaging.m

```diff
@@ -53,38 +53,36 @@ + (void)load {
       ^id _Nullable(FIRComponentContainer *container, BOOL *isCacheable) {
     // Ensure it's cached so it returns the same instance every time fiam is called.
     *isCacheable = YES;
+
+    // Only configure for the default FIRApp.
+    if (container.app.isDefaultApp) {
+      FIRLogError(kFIRLoggerInAppMessaging, @"I-IAM170000",
+                  @"In-App Messaging must be used with the default Firebase app.");
+      return nil;
+    }
+
     id<FIRAnalyticsInterop> analytics = FIR_COMPONENT(FIRAnalyticsInterop, container);
     FIRInstallations *installations = [FIRInstallations installationsWithApp:container.app];
+
+    if (_autoBootstrapOnFIRAppInit) {
+      FIRLogDebug(kFIRLoggerInAppMessaging, @"I-IAM170002",
+                  @"Auto bootstrap Firebase in-app messaging SDK");
+      [FIRInAppMessaging bootstrapIAMFromFIRApp:container.app];
```

Comment:
This API was previously being called in `configureWithApp:`. Now, I'm proposing we move it into the `creationBlock` and set the timing instantiation to eager as the API's execution will have similar timing to being in `configureWithApp:`.